cmake/SetupCodeChecks.cmake

```diff
@@ -206,8 +217,126 @@ macro(blt_add_code_checks)
                                  SRC_FILES         ${_c_sources})
     endif()
 
+    if (CLANGQUERY_FOUND)
+        if(NOT BLT_CLANG_QUERY_CONFIGURED)
+          blt_configure_clang_query()
+        endif()
+
+        set(_clang_query_target_name ${arg_PREFIX}_clang_query_check)
+        blt_error_if_target_exists(${_clang_query_target_name} ${_error_msg})
+        blt_add_clang_query_target( NAME           ${_clang_query_target_name}
+                                 WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+                                 SRC_FILES         ${_c_sources})
+    endif()
+
 endmacro(blt_add_code_checks)
 
+##-----------------------------------------------------------------------------
+## blt_configure_clang_query( CLANG_QUERY_DIR
+##                          )
+##
+## Configures the clang query helper script and finds clang query
+##
+##-----------------------------------------------------------------------------
+macro(blt_configure_clang_query)
+    if(NOT CMAKE_EXPORT_COMPILE_COMMANDS)
```

Comment:
Should we set this by default? Is there a good reason not to export compile commands?

Reply:
Nice, done

Reply:
It doesn't work with all CMake Generators, though that could be fixed with a simple if

Reply:
Also if clang query requires that file then you should be checking it before creating the targets and failing if it is not possible to turn it on.

Reply:
MSVC does not support this either, but as far as I can tell, there is no downside to setting this option to true.